src/tools/build-manifest/src/main.rs

```diff
@@ -375,6 +378,25 @@ impl Builder {
         }
     }
 
+    /// If a tool does not pass its tests, don't ship it.
+    /// Right now, we do this only for Miri.
+    fn check_toolstate(&mut self) {
+        let toolstates: Option<HashMap<String, String>> = try {
+            let toolstates = File::open(self.input.join("toolstates-linux.json")).ok()?;
+            serde_json::from_reader(&toolstates).ok()?
+        };
```

Comment:
Do we need an unstable feature just for that? It can be rewritten to:

```rust
File::open(self.input.join("toolstates-linux.json")).ok()
    .and_then(|toolstates| serde_json::from_reader(&toolstates).ok());
```

Reply:
To expand a bit on why I'd prefer not to use unstable features in this tool: the tool is not ran or tested by CI, but just by `promote-release` on central station. If there is a breaking change in try blocks or they get stabilized the release process would break until we notice nightlies are missing.

Reply:
I just really like `try` blocks. ;)
But sure, I can refactor this.